db/write_thread.cc

```diff
@@ -137,6 +141,43 @@ uint8_t WriteThread::AwaitState(Writer* w, uint8_t goal_mask,
   // 1/sampling_base.
   const int sampling_base = 256;
 
+  if (enable_multi_thread_write_) {
+#ifdef NDEBUG
```

Comment:
what's the reason debug build and release build logic are different?

Reply:
Because unit test need to block and wait SYNC_POINT to wake up. But in product environment, we need it not block thread so that it could do more jobs.

Reply:
Would be great if release build and debug build can use the same logic. Otherwise rocksdb unit tests (maybe also tikv unit tests and  schrodinger tests?) is not testing the real logic.